graph/src/blockchain/block_stream.rs

```diff
@@ -429,9 +429,19 @@ async fn scan_subgraph_triggers<C: Blockchain>(
     }
 }
 
+#[derive(Debug)]
+pub enum EntitySubgraphOperation {
```

Comment:
I am not sure if this makes sense, but might be better to stick the entity into the operation, i.e. to have `Create(Entity)`, `Modify(Entity)`, `Delete(Entity)`; that way, users of this enum are forced to consider each of these cases, and it would make it easy to, e.g., change to `Delete(Id)`, i.e., not load the entire entity if that's useful.

But I don't understand how all of composition fits together, so I am not sure if this suggestions is really an improvement.

Reply:
I guess here the naming is poor. The enum `EntitySubgraphOperation` is a kind of operation more akin to the C style enums. Where the struct just bellow it `EntityWithType` contains the whole 'entity event' with all the fields including this operation kind and the entity itself.

Reply:
I did some renaming which I believe is closer to the semantics. I done it in a separate [PR](https://github.com/graphprotocol/graph-node/pull/5728/files) in order to save on rebasing.